wom/src/main/scala/wom/graph/ExpressionNode.scala

```diff
@@ -1,24 +1,98 @@
 package wom.graph
 
-
+import cats.instances.list._
+import cats.syntax.traverse._
+import cats.syntax.validated._
+import lenthall.Checked
 import lenthall.validation.ErrorOr.ErrorOr
+import lenthall.validation.ErrorOr.ShortCircuitingFlatMap
+import lenthall.validation.Validation._
 import shapeless.Coproduct
-import wom.expression.WomExpression
+import wdl.types.WdlType
+import wdl.values.WdlValue
+import wom.expression.{IoFunctionSet, WomExpression}
 import wom.graph.CallNode.InputDefinitionPointer
-import wom.graph.GraphNodePort.{GraphNodeOutputPort, OutputPort}
-
-final case class ExpressionNode(override val identifier: WomIdentifier, instantiatedExpression: InstantiatedExpression) extends GraphNode {
-
-  val womType = instantiatedExpression.womReturnType
-  val singleExpressionOutputPort = GraphNodeOutputPort(localName, womType, this)
+import wom.graph.GraphNodePort.{ConnectedInputPort, GraphNodeOutputPort, InputPort, OutputPort}
 
+/**
+  * Encapsulates a WomExpression with input ports connected to the expression's dependencies.
+  */
+class ExpressionNode(override val identifier: WomIdentifier,
```

Comment:
`abstract class`?

Reply:
FWIW I'm not sure I like renaming the instantiated version to `DeclarationNode` since that's very WDL specific but 🤷‍♂️

Reply:
Yeah I don't like it either but I couldn't find something better (ExternalNode ?)

Reply:
DeclaredTypeExpressionNode?

Reply:
`ExposedNode` ?

Reply:
🐦-able Opinion: I would have `ExpressionNode` for the super type, rename the "just for call inputs" ones something like `AnonymousExpressionNodes`, and the exposed ones "NamedExpressionNodes" or "ExposedExpressionNodes"